cpp/src/dataframe/cudf_table.cuh

```diff
@@ -518,9 +518,9 @@ public:
     bool operator()(void const * my_column, size_type my_row_index,
                     void const * other_column, size_type other_row_index)
     {
-      col_type const my_elem = static_cast<col_type const*>(my_column)[my_row_index];
-      col_type const other_elem = static_cast<col_type const*>(other_column)[other_row_index];
-      return cudf::detail::unwrap(my_elem) == cudf::detail::unwrap(other_elem);
+      col_type const my_elem{static_cast<col_type const*>(my_column)[my_row_index]};
+      col_type const other_elem{static_cast<col_type const*>(other_column)[other_row_index]};
+      return my_elem == other_elem;
```

Comment:
See my responses to `copy_element`. Short answer is because we're working with type-erased data buffers. 

You could accomplish the same behavior with virtual member functions, but I decided against virtual functions in favor of rolling my own runtime type dispatching with the `type_dispatcher`.

Reply:
Of course virtual methods are inappropriate, but - see my own additional comment on `copy_element`. I still think there should be a way to get rid of these (although not in this PR).